ui/pages/confirmations/components/confirm/footer/footer.tsx

```diff
@@ -228,9 +230,40 @@ const Footer = () => {
       rejectApproval({ location });
       resetTransactionState();
     },
-    [currentConfirmation, isUpgradeTransaction],
+    [
+      currentConfirmation,
+      isUpgradeTransaction,
+      rejectApproval,
+      resetTransactionState,
+    ],
   );
 
+  const onTransactionSubmit = useCallback(() => {
+    const transactionMeta = currentConfirmation as TransactionMeta;
+
+    const finalTransactionMeta: TransactionMeta = {
+      ...transactionMeta,
+      batchTransactions: selectedGasFeeToken
+        ? [selectedGasFeeToken.transferTransaction]
+        : undefined,
+      customNonceValue,
+      txParams: {
+        ...transactionMeta.txParams,
+        ...(selectedGasFeeToken
+          ? {
+              gas: selectedGasFeeToken.gas,
+              maxFeePerGas: selectedGasFeeToken.maxFeePerGas,
+              maxPriorityFeePerGas: selectedGasFeeToken.maxPriorityFeePerGas,
+            }
+          : {}),
+      },
+    };
+
+    ///: BEGIN:ONLY_INCLUDE_IF(build-main,build-beta,build-flask)
+    dispatch(updateAndApproveTx(finalTransactionMeta, true, ''));
+    ///: END:ONLY_INCLUDE_IF
```

Comment:
Nit: Above is relatiely bigger function to be included into a component, I would prefer to make this a hook / util.